indexing-service/src/main/java/org/apache/druid/indexing/common/task/Task.java

```diff
@@ -164,7 +164,10 @@ default int getPriority()
    * Asks a task to arrange for its "run" method to exit promptly. Tasks that take too long to stop gracefully will be
    * terminated with extreme prejudice.
    *
-   * This method can be called at any time while {@link #run} is being called when the task is killed.
+   * This method can be called at any time while {@link #run} is being called when the task is killed. If this task
+   * is not started yet, that is {@link #run} is not called yet, this method will be never called.
+   * Once this task is started, this method can be called even after {@link #run} returns. Implementations of this
+   * method may want to avoid unnecessary work if {@link #run} already returned.
    * Depending on the task executor type, one of the two cases below can happen when the task is killed.
```

Comment:
It would be clearer if there was an empty line above this line and no empty line below this line.